arch/riscv/riscv_features.c

```
@@ -1,25 +1,84 @@
+#  endif
+            break;
+        }
+    }
+
+    return 1;
+#else
+    return 0;
+#endif
+}
+
+int Z_INTERNAL riscv_check_features_runtime_hwcap(struct riscv_cpu_features *features) {
```

Cppcheck (2.19.0)

[style] The function 'riscv_check_features_runtime_hwcap' should have static linkage since it is not used outside of its translation unit.

(staticFunction)